extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPusherConfig.java

```diff
@@ -25,6 +25,8 @@
  */
 public class S3DataSegmentPusherConfig
 {
+  public static final int MAX_LISTING_LENGTH = 1000;
```

Comment:
why not having this check at the config it self ? plus i find it ambiguous to allow user set values < 0 and replace it under the hood. IMHO i would add this check at the config class and the call to the `onfig.getMaxListingLength()` must return an acceptable value.